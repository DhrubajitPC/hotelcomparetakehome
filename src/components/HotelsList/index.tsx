import React, { FC } from "react";
import { useAppContext } from "../../state/Context";
import HotelInfo from "../HotelInfo";

const HotelsList: FC = () => {
  const { state } = useAppContext();
  return (
    <>
      {!state.loading.hotel &&
        state.hotels.map((h, idx) => (
          <HotelInfo
            onBook={() => {}}
            currency={""}
            {...h}
            price={212}
            key={idx}
          />
        ))}
    </>
  );
};

export default HotelsList;
