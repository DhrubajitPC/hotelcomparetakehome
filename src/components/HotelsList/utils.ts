import { Hotel, Price } from "../../state/types";

export const orderHotelsbyPriceAvailability = (
  hotels: Hotel[],
  prices: Price[]
) => {
  const hotelsWithPrices = hotels.filter((h) =>
    Boolean(prices.find((p) => p.id === h.id)?.price)
  );
  const hotelsWithoutPrices = hotels.filter(
    (h) => !Boolean(prices.find((p) => p.id === h.id)?.price)
  );

  return [...hotelsWithPrices, ...hotelsWithoutPrices];
};
