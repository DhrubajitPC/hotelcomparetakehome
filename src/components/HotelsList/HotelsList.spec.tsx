import { render } from "@testing-library/react";
import React from "react";
import HotelsList from ".";

const mockContext = {
  state: {
    hotels: [
      {
        id: 1,
        name: "mock hotel name",
        rating: 3,
        stars: 4,
        address: "mock address",
        photo: "image url",
        description: "mock description",
      },
    ],
    loading: { hotel: false, pricing: false },
    selectedCurrency: "USD",
  },
};

jest.mock("../../state/context.tsx", () => ({
  __esmodule: true,
  useAppContext: () => mockContext,
}));

describe("HotelInfo", () => {
  beforeAll(() => {});
  it("should render correctly", () => {
    const { container } = render(<HotelsList />);
    expect(container).toMatchSnapshot();
  });
});
