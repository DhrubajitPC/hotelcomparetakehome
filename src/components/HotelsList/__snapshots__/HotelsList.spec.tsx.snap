// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelInfo should render correctly 1`] = `
<div>
  <div
    style="display: flex; justify-content: left;"
  >
    <div>
      <img
        alt="hotel"
        src="image url"
      />
    </div>
    <div
      style="margin-right: auto;"
    >
      <h3>
        mock hotel name
      </h3>
      <p>
        mock address
      </p>
      <div>
        mock description
      </div>
      <button>
        Book!
      </button>
    </div>
    <div>
      <p>
        4
         star Hotel
      </p>
      <p>
        Rating 
        3
      </p>
      <div
        data-testid="tooltip-container"
        style="position: relative;"
      >
        <p>
          
           
          212
        </p>
      </div>
    </div>
  </div>
</div>
`;
